addons/udes_security/models/models.py

```diff
@@ -74,6 +93,15 @@ def _set_fields_readonly(doc, res):
 
             res["arch"] = etree.tostring(doc)
             # Remove archive option in tree view and make calendar readonly
-            res["readonly_desktop_user"] = True
+            res["readonly_desktop_user"] = True  # Used in basic_view.js
+
+        can_archive_record = ResArchivingRestrictions._get_can_archive(model)
```

Comment:
Can we `else` this with `if self.env.user.u_desktop_readonly and not ModelException.is_exempt(model):`?

If a user is readonly then they're not going to be able to archive anything anyway so the checks here aren't needed.